everestpy/src/everest/framework/error/__init__.pyi

```diff
@@ -0,0 +1,97 @@
+from enum import Enum
+from typing import Callable, overload
+
+class Severity(Enum):
+    Low = "Low",
+    Medium = "Medium",
+    High = "High"
+
+class ImplementationIdentifier:
+    def __init__(self, module_id: str, implementation_id: str) -> None: ...
+
+    @property
+    def module_id(self) -> str: ...
+
+    @property
+    def implementation_id(self) -> str: ...
+
+class UUID:
+    @overload
+    def __init__(self) -> None: ...
+
+    @overload
+    def __init__(self, uuid: str) -> None: ...
+
+class State(Enum):
+    Active = "Active",
+    ClearedByModule = "ClearedByModule",
+    ClearedByReboot = "ClearedByReboot"
+
+class Error:
+    def __init__(self) -> None: ...
+
+    @property
+    def type(self) -> str: ...
+
+    @property
+    def sub_type(self) -> str: ...
+
+    @property
+    def description(self) -> str: ...
+
+    @property
+    def message(self) -> str: ...
+
+    @property
+    def severity(self) -> Severity: ...
+
+    @property
+    def origin(self) -> ImplementationIdentifier: ...
+
+    @property
+    def timestamp(self) -> int: ...
+
+    @property
+    def uuid(self) -> UUID: ...
+
+    @property
+    def state(self) -> State: ...
+
+ErrorCallback = Callable[[Error], None]
+
+class ErrorStateCondition:
+    def __init__(self, type: str, sub_type: str, active: bool) -> None: ...
+
+    @property
+    def type(self) -> str: ...
+
+    @property
+    def sub_type(self) -> str: ...
+
+    @property
+    def active(self) -> bool: ...
+
+class ErrorStateMonitor:
+    def is_error_active(self, type: str, sub_type: str) -> bool: ...
+
+    @overload
+    def is_condition_satisfied(self, condition: ErrorStateCondition) -> bool: ...
+
+    @overload
+    def is_condition_satisfied(self, condition: list[ErrorStateCondition]) -> bool: ...
+
+class ErrorFactory:
+    @overload
+    def create_error(self) -> Error: ...
+
+    @overload
+    def create_error(self, type: str, sub_type: str, message: str) -> Error: ...
```

Comment:
calling this "type" leads to warnings in the test code in everest-core, maybe calling it error_type would be clearer anyway but it's not really a big issue in my view

Reply:
Not sure how to do it with pybind, since this directly binds to the method in `ErrorFactory`. Not sure whether it would cause issues renaming it for python

Reply:
ok, then I would say leave it as is for now and address this if there's every any issues with it in some of the python module code